common/src/test/java/com/bakdata/quick/common/api/client/PartitionedMirrorClientTest.java

```diff
@@ -0,0 +1,167 @@
+        // the underlying PartitionFinder in PartitionRouter.findHost function.
+        this.partitionFinder.enqueue(3);
+        final TopicData topic2 = this.topicDataClient.fetchValue(DEFAULT_TOPIC);
+        // If the info had not been updated, we would have received IllegalStateException because
+        // the partitionToMirrorHost would not have access to the partition=3.
+        assertThat(Objects.requireNonNull(topic).getName()).isEqualTo(DEFAULT_TOPIC);
+        assertThat(Objects.requireNonNull(topic2).getName()).isEqualTo(DEFAULT_TOPIC);
+    }
+
+    @Test
+    void shouldReadHeaderAndUpdateRouterWithFewerReplicas() throws JsonProcessingException {
+        // Second response: A dummy response for PartitionedMirrorClient from Mirror contains the X-Cache-Update header.
+        // The header is set to simulate a mapping change from partition to host.
+        final TopicData topicData = createTopicData();
+        final String body = TestUtils.generateBody(topicData);
+        this.server.enqueue(new MockResponse().setBody(body).setHeader(
+            HeaderConstants.getCacheMissHeaderName(), HeaderConstants.getCacheMissHeaderValue()));
+
+        // Third response: A new mapping for PartitionedMirrorClient
+        final String routerBody = TestUtils.generateBodyForRouterWith(Map.of(1, this.host));
+        this.server.enqueue(new MockResponse().setBody(routerBody));
+
+        // The PartitionedRouter of PartitionedMirrorClient will get partition=2 from
+        // the underlying PartitionFinder in PartitionRouter.findHost function.
+        this.partitionFinder.enqueue(2);
+        // In the current scenario, we will make two calls to Mirror when we fetch a value.
+        // The first is done to get a value (second response), second to get mapping for updating PartitionRouter
+        // (third response)
+        final TopicData topic = this.topicDataClient.fetchValue(DEFAULT_TOPIC);
+        // Fourth response: A dummy message for yet another call to Mirror is needed
+        // to test whether the mapping has been successfully updated.
+        this.server.enqueue(new MockResponse().setBody(body));
+        // The PartitionedRouter of PartitionedMirrorClient will get partition=3 from
+        // the underlying PartitionFinder in PartitionRouter.findHost function.
+        this.partitionFinder.enqueue(1);
+        final TopicData topic2 = this.topicDataClient.fetchValue(DEFAULT_TOPIC);
+        // If the info had not been updated, we would have received IllegalStateException because
+        // the partitionToMirrorHost would not have access to the partition=3.
+        assertThat(Objects.requireNonNull(topic).getName()).isEqualTo(DEFAULT_TOPIC);
+        assertThat(Objects.requireNonNull(topic2).getName()).isEqualTo(DEFAULT_TOPIC);
+
+        this.partitionFinder.enqueue(2);
+        // Since there was no mapping update, an exception will be thrown.
+        // There is no host for partition=3
```

Comment:
Why partition 3? Isn't this about partition 2?

```suggestion
        // There is no host for partition=2
```

No?